exporter/exporterhelper/batch_sender.go

```diff
@@ -0,0 +1,461 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterhelper // import "go.opentelemetry.io/collector/exporter/exporterhelper"
+
+import (
+	"context"
+	"errors"
+	"runtime"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+)
+
+// MergeBatcherConfig defines a basic configuration for batching requests based on a timeout and a minimum number of
+// items.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
```

Comment:
IMO we should start designing things with exporterhelper with 1.0 in mind. This doesn't mean that it will happen any time soon, but at least we should not make it harder to publish exporterhelper 1.0.

To this end, I think we should publish this functionality as a separate module (`exporter/exporterhelper/batcher`). This needs a bit of a refactor to be able to define `WithBatcher` on a separate module (we likely need an exporter/exporterhelper/internal` module to be able to have options), but allows us to eventually declare `exporter/exporterhelper` as 1.0 without declaring this (or other functionality) as part of 1.0.

What do you think? I can try and do this for other experimental functionality on other modules and see how it works out, but here I think it should be doable.

Reply:
I agree, I was planning to move it to a sub-package, but I'm not sure if `exporter/exporterhelper/batcher` deserves to be a module. I think we can wait for the transition and release exporter 1.0

Reply:
:+1: Package is fine, we can do module in the future if we want to release 1.0 before we stabilize this

Reply:
Given that we already have a bunch of API related to the exporterhelper v2 in this module, I would leave this here just for now, marked as experimental the same way as the existing experimental API. We can move this out before 1.0 or agree to have `experimental` API parts in 1.x. The problem is that we have to move `Request` API from here to be able to move the batcher API. We cannot agree with @bogdandrutu to move the `Request` API yet :) See https://github.com/open-telemetry/opentelemetry-collector/pull/8867#discussion_r1391802064

Reply:
Okay, makes sense, no need to do this right now, let's continue with the status quo of doing things and change after we have general rules for how to add experimental APIs